docs/controllers/leaderelection.md

```diff
@@ -0,0 +1,62 @@
+# Leader Election
+
+This chapter is a stub about Leases, Leader Election and how to achieve distributed locks around Kubernetes resources.
+
+!!! note "Terminology"
+
+    See [Kubernetes//Leases](https://kubernetes.io/docs/concepts/architecture/leases/) for introductory definitions.
+
+
+## Crates
+
+At the moment, leader election support is not exported from `kube` direct, and requires 3rd party crates (as decided in [kube#485](https://github.com/kube-rs/kube/issues/485#issuecomment-1837386565)).
+
+### kube-leader-election
+The [`kube-leader-election` crate](https://crates.io/crates/kube-leader-election/) via [hendrikmaus/kube-leader-election](https://github.com/hendrikmaus/kube-leader-election) implements a simple and low-level form of leader election with a [use-case disclaimer](https://github.com/hendrikmaus/kube-leader-election?tab=readme-ov-file#kubernetes-lease-locking).
+
+```rust
+use kube_leader_election::{LeaseLock, LeaseLockParams};
+let leadership = LeaseLock::new(client, namespace, LeaseLockParams { ... });
+let _lease = leadership.try_acquire_or_renew().await?;
+leadership.step_down().await?;
+```
```

Comment:
These are fairly vague terms, what does "simple and low-level" mean compared to "high-level and sophisticated"?

That kind of language makes me wonder if we're adding much with these summaries compared to just having a flat list of links and letting each implementation speak for itself.

Reply:
There's also a fine line to walk between "here are some options we've heard of in the past" and outright endorsements.

Reply:
> vague terms

yeah, that is fair. i was trying to avoid making strong value judgements while distinguishing between the options (which do have varying degrees of closeness to the underlying api). i'll cut the sophisticated at the very least, that's harder to intuit and also potentially constitutes more of a value-judgement (which it wasn't intended to have).

> fine line to walk

yeah, it's true. i have never tested these, but they all do seem fine on the surface from a quick library evaluation. maybe worth clarifying.

Reply:
it's now an entirely flat list as suggested